### PR TITLE
Fix: convert header to functional component

### PR DESCRIPTION
**Pull request by akankshat05**
Fixes [T348415](https://phabricator.wikimedia.org/T348415)
## Proposed Changes
- Header into a functional component and used the useState hook
## Files Updated
- `header.js`

### CODE REVIEW

File: components/Header.js [hunk 1]
```diff
@@ -1,114 +1,105 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import Link from "next/link";
 import { withSession } from "../hooks/withSession";
```
Comment (Parthiv-M) [inline components/Header.js:3]: Please remove this unused import statement

**Comment by Parthiv-M** [comment]
You might want to wait until #182 is merged and then update the `Header` component? Rest looks alright to me.

File: components/Header.js (hunk 1 above)
Comment (akankshat05) [reply components/Header.js]: I have removed it.

**Review by Parthiv-M — APPROVED** [review]
Works fine for me. 
Since other changes in the header are stalled for now, we can get this merged

**Pull request merged**